src/changelog-config.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable node/no-process-env, node/no-sync */
+
+import fs from 'fs';
+import path from 'path';
+
+type PackageJson = {
+  autoChangelog: {
+    packageRename: {
+      originalLastestVersion: string;
+      originalTagPrefix: string;
+    };
+  };
+};
+
+/**
+ * Returns the latest version of the original package in case of package renamed.
+ *
+ * @returns The original package latest version.
+ */
+export function getOriginalLatestVersion(): string | null {
+  // Set automatically by Yarn 3.x
+  const packageJsonPath = process.env.npm_package_json;
+  if (packageJsonPath) {
+    const packageJson = path.resolve(packageJsonPath);
+    const packageJsonContent = JSON.parse(
+      fs.readFileSync(packageJson, 'utf8'),
+    ) as PackageJson;
+
+    if (
+      typeof packageJsonContent.autoChangelog?.packageRename
+        ?.originalLastestVersion === 'string'
+    ) {
+      return packageJsonContent.autoChangelog.packageRename
+        .originalLastestVersion;
+    }
+  }
+
+  return null;
+}
+
+/**
+ * Returns the original package tag prefix in case of package renamed.
+ *
+ * @returns The original package tag prefix.
+ */
+export function getOriginalTagPrefix(): string | null {
+  // Set automatically by Yarn 3.x
+  const packageJsonPath = process.env.npm_package_json;
+  if (packageJsonPath) {
+    const packageJson = path.resolve(packageJsonPath);
+    const packageJsonContent = JSON.parse(
+      fs.readFileSync(packageJson, 'utf8'),
+    ) as PackageJson;
+    if (
+      typeof packageJsonContent.autoChangelog?.packageRename
+        ?.originalTagPrefix === 'string'
+    ) {
+      return packageJsonContent.autoChangelog.packageRename.originalTagPrefix;
+    }
+  }
+
+  return null;
+}
```

Comment:
A couple changes here might reduce the complexity of the logic and add a bit of type safety:
- Early exit in case of `undefined` `packageJsonPath` -> avoids unnecessary nesting.
- Replacing `as` type assertion for with type annotation.
- By removing the `as` assertion and typing `packageJsonContent` as `PackageJson`, we no longer need to check whether `originalTagPrefix` is not a string, since if that happens it will trigger a "not assignable" type error on line 54.

```suggestion
  if (packageJsonPath === undefined) {
    return null;
  }

  const packageJson = path.resolve(packageJsonPath);
  const packageJsonContent: PackageJson = JSON.parse(
    fs.readFileSync(packageJson, 'utf8'),
  );

  return (
    packageJsonContent.autoChangelog?.packageRename?.originalTagPrefix ?? null
  );
```